CMakeLists.txt

```diff
@@ -40,6 +40,10 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "17|20")
   message(FATAL_ERROR "Unsupported C++ standard: ${CMAKE_CXX_STANDARD}")
 endif()
 
+if(CMAKE_CXX_STANDARD EQUAL 17)
+  message(WARNING "C++17 is deprecated and support for it will be removed in the future. Please use at least C++20.")
```

Comment:
Should we put a version here when we are going to remove it? If we want to have a tag in between where this message should be visible that would make it v01-03.

Reply:
Added as discussed today in the podio and EDM4hep meeting.